### ISSUE TITLE
There is no abilitity to add a tab to product page at a desired position

### ISSUE THREAD

**Issue by stasleo** [magento/magento2#4154]
### Preconditions
Magento 2.2

### Steps to reproduce
Try to change the position of a tab on a product page in the code

### Expected result
Obviously, most store owners would like to have an ability to change the order of any tab on a product page. For example, they would prefer to make "Description" tab appearing first always, regardless names of modules that add tabs to a product page.

### Actual result
There is no normal way to change an order of tabs on Magento 2 storefront on a product page. Magento2 just loads tabs from A to Z by its module name. So, if there is a module with A_modulename name, then its tab will appear first (before the Description tab).

**Comment by devhn** [comment]
Same issue here - we also tried repositioning with after/before other tabs within the xml and it's not working either.

**Comment by stasleo** [comment]
any updates on this?

**Comment by MomotenkoNatalia** [comment]
Hi @stasleo !
Thank you for reporting this issue. Internal MAGETWO-53105 ticket has been created.

**Comment by davidnguyen2205** [comment]
I got the temporary solution that use sequence module in your xml:

```
<module name='Your_Module' setup_version='2.0.0'>
    <sequence>
         <!-- Your tab will be added on the last place -->
         <module name='Magento_Review'/>
    </sequence>
</module>
```

**Comment by jarhody** [comment]
Anything new on this? Kind of a bummer.

**Comment by piotrekkaminski** [comment]
Thank you for your submission.

We recently made some changes to the way we process GitHub submissions to more quickly identify and respond to core code issues.

Feature Requests and Improvements should now be submitted to the new [Magento 2 Feature Requests and Improvements](https://community.magento.com/t5/Magento-2-Feature-Requests-and/idb-p/feature-requests) forum (see details [here](https://community.magento.com/t5/News-Announcements/Improvements-to-GitHub-Management/m-p/44572#M96)).

We are closing this GitHub ticket and have moved your request to the new [forum](https://community.magento.com/t5/Magento-2-Feature-Requests-and/There-is-no-abilitity-to-add-a-tab-to-product-page-at-a-desired/idi-p/46023).

**Comment by davemacaulay** [comment]
I don't believe this is a new feature request and is indeed a bug, can we re-open please.

I believe this is a bug as the block & template that drives the product descriptions should respect the layout ordering (`after` & `before`) but in actual fact it just renders the tabs in the order their respected modules are loaded. This means plugins or _hacks_ are required to re-order the tabs, even though they're controlled by the layout system.

The implementation of tabs are as follows within `catalog_product_view.xml`:
```
<block class="Magento\Catalog\Block\Product\View\Description" name="product.info.details" template="Magento_Catalog::product/view/details.phtml" after="product.info.media">
    <block class="Magento\Catalog\Block\Product\View\Description" name="product.info.description" as="description" template="Magento_Catalog::product/view/attribute.phtml" group="detailed_info">
        <arguments>
            <argument name="at_call" xsi:type="string">getDescription</argument>
            <argument name="at_code" xsi:type="string">description</argument>
            <argument name="css_class" xsi:type="string">description</argument>
            <argument name="at_label" xsi:type="string">none</argument>
            <argument name="title" translate="true" xsi:type="string">Details</argument>
        </arguments>
    </block>
    <block class="Magento\Catalog\Block\Product\View\Attributes" name="product.attributes" as="additional" template="Magento_Catalog::product/view/attributes.phtml" group="detailed_info">
        <arguments>
            <argument translate="true" name="title" xsi:type="string">More Information</argument>
        </arguments>
    </block>
</block>
```
Any block added within the `product.info.details` block will not respect the layout XML order.

**Comment by ghost** [comment]
@stasleo, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by magento-engcom-team** [comment]
Hi @vasilii-b. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by magento-engcom-team** [comment]
Hi @stasleo. Thank you for your report. 
The issue has been fixed in magento/magento2#19007 by @vasilii-b in 2.3-develop branch
Related commit(s):
 - [c91649a8ef2e7f9d50aedead338537afd7b89401](https://github.com/magento/magento2/commit/c91649a8ef2e7f9d50aedead338537afd7b89401)
 - [c1d0f31d6be809083f2dc31a73ea52eae1f740b8](https://github.com/magento/magento2/commit/c1d0f31d6be809083f2dc31a73ea52eae1f740b8)
 - [1ad7d4598504fe89b2f0899a966f61fbdf5c64d5](https://github.com/magento/magento2/commit/1ad7d4598504fe89b2f0899a966f61fbdf5c64d5)

The fix will be available with the upcoming 2.3.1 release.
